sql/core/src/test/scala/org/apache/spark/sql/connector/KeyGroupedPartitioningSuite.scala

```diff
@@ -1039,4 +1039,60 @@ class KeyGroupedPartitioningSuite extends DistributionAndOrderingSuiteBase {
       }
     }
   }
+
+  test("SPARK-44641: duplicated records when SPJ is not triggered") {
```

Comment:
May I ask which Spark versions are affected, @sunchao ? (Also, cc @aokolnychyi )

Reply:
@dongjoon-hyun this affects Spark 3.4.0 and 3.4.1

Reply:
it'd be great if we can get this in Spark 3.5.0

Reply:
+1 for backporting to branch-3.5.

BTW, could you make it as `Blocker` to achieve that?

Reply:
Sure, it's done